src/data/int/cast.lean

```diff
@@ -209,16 +209,54 @@ namespace monoid_hom
 variables {M : Type*} [monoid M]
 open multiplicative
 
-theorem ext_int {f g : multiplicative ℤ →* M}
+theorem ext_mint {f g : multiplicative ℤ →* M}
   (h1 : f (of_add 1) = g (of_add 1)) : f = g :=
 begin
   ext,
   exact add_monoid_hom.ext_iff.1
     (@add_monoid_hom.ext_int _ _ f.to_additive g.to_additive h1) _,
 end
 
+/-- If two `monoid_hom`s agree on `-1` and the naturals then they are equal. -/
+theorem ext_int {f g : ℤ →* M}
+  (h_neg_one : f (-1) = g (-1))
+  (h_nat : f.comp int.of_nat_hom.to_monoid_hom = g.comp int.of_nat_hom.to_monoid_hom) :
+  f = g :=
+begin
+  ext (x | x),
+  { exact (monoid_hom.congr_fun h_nat x : _), },
+  { rw [int.neg_succ_of_nat_eq, ← neg_one_mul, f.map_mul, g.map_mul],
+    congr' 1,
+    exact_mod_cast (monoid_hom.congr_fun h_nat (x + 1) : _), }
+end
+
 end monoid_hom
 
+namespace monoid_with_zero_hom
+
+variables {M : Type*} [monoid_with_zero M]
+
+/-- If two `monoid_with_zero_hom`s agree on `-1` and the naturals then they are equal. -/
+theorem ext_int {f g : monoid_with_zero_hom ℤ M}
+  (h_neg_one : f (-1) = g (-1))
+  (h_nat : f.comp int.of_nat_hom.to_monoid_with_zero_hom =
+           g.comp int.of_nat_hom.to_monoid_with_zero_hom) :
+  f = g :=
+-- TODO: `monoid_with_zero_hom.to_monoid_hom_injective`
+suffices f.to_monoid_hom = g.to_monoid_hom,
+by {
+  rw monoid_hom.ext_iff at this,
+  ext,
+  simpa using this x },
```

Comment:
The style I generally see is:
```suggestion
suffices f.to_monoid_hom = g.to_monoid_hom,
{ rw monoid_hom.ext_iff at this,
  ext,
  simpa using this x },
```
Is is also common to write it the way it currently is?

Reply:
This is not in a tactic block, so that's not legal! Of course, really we should just add the `monoid_with_zero_hom.to_monoid_hom_injective` lemmas. If you're looking for an easy PR to make...

Reply:
Ah, I didn't catch that!